compiler-core/src/main/java/toothpick/compiler/common/ToothpickOptions.kt

```diff
@@ -79,6 +80,14 @@ data class ToothpickOptions(
          * Might be useful for debugging or test purposes.
          */
         const val VerboseLogging = "toothpick_verbose_log"
+
+        /**
+         * The name of the annotation processor option to use `with` control flow
+         * to avoid `NAME_SHADOWING` issue.
+         *
+         * Might be useful for debugging or test purposes.
+         */
+        const val WrapScopeIntoWithControlFlow = "toothpick_wrap_scope_into_with_control_flow"
```

Comment:
Is it really necessary to make this an option? Is there a downside to systematically use a `with` control flow to handle the scope? If we can avoid handling the two systems, we might as well, it would be much easier to code, test and review

Reply:
Removed it completely and changed `WrapScopeIntoWithControlFlowTest` to `ParametrizedParametersInjectionTest`, as it still checks an injection of generic parameters for cases when injected type is not strictly equals provided.